### DISCUSSION TITLE
global-ray-loader.php required from unkown in line 0.

### DISCUSSION THREAD

**Discussion by KarlisJ** [spatie/global-ray#27] [Q&A] [answered]
I can't be 100% confident about the order of actions taken by me but the big picture is that I manually removed global-ray from the global composer `composer.json` file and reinstalled the dependencies. 

I don't have ray installed anymore, but whenever I try to execute composer (PHP in general), I get the following error
> Warning: Unknown: failed to open stream: No such file or directory in Unknown on line 0
>
> Fatal error: Unknown: Failed opening required '/Users/karlisjanisels/.composer/vendor/spatie/global-ray/src/scripts/global-ray-loader.php' (include_path='.:/opt/homebrew/Cellar/php@7.4/7.4.30_1/share/php@7.4/pear') in Unknown on line 0

I've tried reinstalling composer, fully removing all caches, composer dirs, but whenever I install composer again I experience the same error. 

I can't reinstall the ray package because the composer is throwing the error. `in Unknown on line 0` also doesn't tell much :) 

Could someone help me understand where the file is required so that I could understand how to clean up the composer?

**Comment by freekmurze** [comment, marked as answer]
Have you tried this?
https://github.com/spatie/global-ray#troubleshooting

**Reply by KarlisJ** [reply]
Thank you, removing `auto_prepend_file` directive fixed the issue.
